src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -107,7 +109,13 @@ public void Invoke(IDictionary<string, string> argsDictionary)
                 {
                     var dataCollectionTestCaseEventSender = DataCollectionTestCaseEventSender.Create();
                     dataCollectionTestCaseEventSender.InitializeCommunication(dcPort);
-                    dataCollectionTestCaseEventSender.WaitForRequestSenderConnection(ClientListenTimeOut);
+
+                    // It's possible that connection to vstest.console happens, but to datacollector fails, why?
+                    // DataCollector keeps the server alive for testhost only for 15secs(increased to 60 now), 
+                    // if somehow(on slower machines, with Profiler Enabled) testhost can take considerable time to launch,
+                    // in such scenario dc.exe would have killed the server, but testhost will wait infinitely to connect to it,
+                    // hence do not wait to connect to datacollector process infinitely, as it will cause process hang.
+                    dataCollectionTestCaseEventSender.WaitForRequestSenderConnection(DataConnectionClientListenTimeOut);
```

Comment:
>WaitForRequestSenderConnection [](start = 54, length = 30)

if we fail to connect we aren't logging anything .. minimal we should do that.. also what will happen when timeout kicks in,,,

Reply:
Logging done. If timeout kicks in vstest will only run test, & data collection will fail